api/repositories/domain_repository.go

```diff
@@ -38,18 +46,36 @@ type ListDomainsMessage struct {
 }
 
 func (r *DomainRepo) GetDomain(ctx context.Context, authInfo authorization.Info, domainGUID string) (DomainRecord, error) {
-	domain := &networkingv1alpha1.CFDomain{}
-	err := r.privilegedClient.Get(ctx, types.NamespacedName{Name: domainGUID}, domain)
+	domainList := &networkingv1alpha1.CFDomainList{}
+	err := r.privilegedClient.List(ctx, domainList, client.MatchingFields{"metadata.name": domainGUID})
 	if err != nil {
-		switch {
-		case k8serrors.IsNotFound(err):
-			return DomainRecord{}, NewNotFoundError("Domain", err)
-		default:
-			return DomainRecord{}, fmt.Errorf("get-domain: k8s get failed: %w", err)
-		}
+		return DomainRecord{}, fmt.Errorf("get-domain: privileged list failed: %w", err)
+	}
+
+	if len(domainList.Items) == 0 {
+		return DomainRecord{}, NewNotFoundError("Domain", err)
+	}
+	if len(domainList.Items) > 1 {
+		return DomainRecord{}, errors.New("get-domain duplicate domains exist")
+	}
+
+	domain := cfDomainToDomainRecord(&domainList.Items[0])
+
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return DomainRecord{}, fmt.Errorf("get-domain failed to build user client: %w", err)
+	}
+
+	err = userClient.Get(ctx, client.ObjectKey{Namespace: domain.Namespace, Name: domain.GUID}, &networkingv1alpha1.CFDomain{})
+	if k8serrors.IsForbidden(err) {
+		return DomainRecord{}, NewForbiddenError(err)
+	}
+
+	if err != nil { // untested
+		return DomainRecord{}, fmt.Errorf("get-domain user client get failed: %w", err)
 	}
 
-	return cfDomainToDomainRecord(domain), nil
+	return domain, nil
```

Comment:
I would recommend returning a record for the domain returned by the unprivileged client. Not sure whether that would make any difference, but I am a bit worried about returning stuff the user is not supposed to see.

Reply:
> I can't foresee a situation where they would get a result they don't have access to
Yeah, it's kinda just to help us from our future selves IMO. I put some similar thoughts on this PR:
https://github.com/cloudfoundry/cf-k8s-controllers/pull/634#discussion_r804800434

I think it would be nice to move the privileged client interactions into a private function that gives some more context to why we're doing that (to look up a namespace essentially).